### ISSUE TITLE
pbench-fio fails to run the test even if fio is installed in correct version

### ISSUE THREAD

**Issue by jhutar** [distributed-system-analysis/pbench#2449]
I have pbench-agent and fio installed (I had older fio version, had to `dnf upgrade` to a version prefered by pbench-agent):

```
[root@... ~]# rpm -q pbench-agent fio
pbench-agent-0.69.9-1ga12a9aa0f.noarch
fio-3.21-6.pbench.x86_64
```

but it is failing:

```
[root@... ~]# pbench-fio -d /var/lib/kafka/ --samples=1
[...]
running fio job: /var/lib/pbench-agent/fio__2021.09.07T07.39.38/1-read-4KiB/fio.job (sample1)
/var/lib/pbench-agent/fio__2021.09.07T07.39.38/1-read-4KiB/sample1/fio.cmd: line 1: /usr/local/bin/fio: No such file or directory
[...]
```

To make it work, I had to create  symlink like this:

```
[root@... ~]# ln -s /usr/bin/fio /usr/local/bin/fio
```

**Comment by jhutar** [comment]
Also Peter asked me to provide:

```
[root@e26-h27-740xd ~]# yum list installed pbench-agent fio pbench-fio | tee
Updating Subscription Management repositories.
Unable to read consumer identity

This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.

Installed Packages
fio.x86_64          3.21-6.pbench      @copr:copr.fedorainfracloud.org:ndokos:pbench
pbench-agent.noarch 0.69.9-1ga12a9aa0f @copr:copr.fedorainfracloud.org:ndokos:pbench
```

**Comment by bengland2** [comment]
@jhutar was that all you had to do?   If so, perhaps we should just have pbench-fio shell script use the executable "fio" instead of a full pathname and let PATH variable direct us to the right fio executable.  There was a pbench-fio RPM package at one point provided by the pbench team, but I suspect this is no longer needed and you are better off using either fio package provided with RHEL8 (in /usr/bin/fio) or upstream fio executable (probably installed in /usr/local/bin/).   Comments anyone?

**Comment by jhutar** [comment]
Hello Ben. For setup, I followed https://distributed-system-analysis.github.io/pbench/start.html#install and https://distributed-system-analysis.github.io/pbench/start.html#steps and recorded it into a playbook, so these are the steps:

    - name: "Enable EPEL"
      ansible.builtin.package:
        name: https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm
        state: installed

    - name: "Enable Code Ready Builder"
      yum_repository:
        name: crd
        description: Code Ready Builder
        gpgcheck: no
        baseurl: http://.../content/dist/rhel8/8/x86_64/codeready-builder/os/

    - name: "Setup pbench repo file"
      get_url:
        url: https://copr.fedorainfracloud.org/coprs/ndokos/pbench/repo/epel-8/ndokos-pbench-epel-8.repo
        dest: /etc/yum.repos.d/ndokos-pbench-epel-8.repo

    - name: "Install pbench-agent"
      package:
        name:
          - pbench-agent
          - pbench-sysstat
        state: installed

    - name: "Deploy pbench-agent ssh key"
      get_url:
        url: "http://.../git/pbench.git/plain/agent/production/ssh/id_rsa"
        dest: "/opt/pbench-agent/"
        mode: "0600"

    - name: "Deploy pbench-agent config"
      get_url:
        url: "http://.../git/pbench.git/plain/agent/production/config/pbench-agent.cfg"
        dest: "/opt/pbench-agent/config/"
        mode: "0644"

Then I have just run the `pbench-fio` command, it complained about version, so I `dnf -y upgrade fio` and then I got what is reported above.

**Comment by bengland2** [comment]
@jhutar upstream fio is now up to v3.27 - if the pbench team is going to package the fio benchmark, let's make it a more up-to-date version.   The original reason for a pbench-fio RPM, as I recall, was that there were experimental features in pbench-fio that were not yet in upstream fio (histogram logging) 4-5 years ago.   All those changes were merged upstream a long time ago.

**Comment by jhutar** [comment]
@bengland2 I see. I do not have a preference about exact fio version. I only reported this bug that even if I followed guide, the benchmark command failed.

**Comment by portante** [comment]
First, let me respond to address the bug as reported:  The `v0.69.9` agent does not work with the normal `fio` RPMs where they install the binary in `/usr/bin`.  This was fixed by @ldoktor in https://github.com/distributed-system-analysis/pbench/commit/b8cb2b81f182dfc47630a35fed2de84798628e1c, about a year and half ago, but the fix was not back-ported to the `b0.69` branch.

The work-around of using a symlink is probably the best option until we can either release the v0.71 pbench-agent RPM, or back-port this above commit.

Regarding `fio` and who provides RPM packages for it ... @bengland2, the `pbench-fio` interface has been modified quite a while ago to work with at least version `3.21` of `fio` and later.  Our `pbench-fio` interface no longer has an exact version requirement.

While we don't announce it publicly as being associated with the `pbench` GitHub project, we do gratuitously provide a `3.21-6.pbench` package the `fio` benchmark for versions which we minimally support on platforms where it is provided by default but with an older version we don't support (e.g. RHEL/CentOS 7).  See @ndokos's COPR repository: https://copr.fedorainfracloud.org/coprs/ndokos/pbench/packages/

The latest `fio` is shipped with Fedora (33/34 at the time this was written), and RHEL / CentOS ships a `fio` it maintains, since it is part of those products.

So if you need a version of `fio` for a platform not provided, we can either help you build it yourself, or work to get that version published through official channels.

We don't have the bandwidth, nor the know how, maintain our own up-to-date version of `fio` for all platforms we support.

**Comment by portante** [comment]
Fixed by #2519.
